### PR TITLE
AG-12978 Fix stacked normalised area with missing x values

### PR DESCRIPTION
**Pull request by lsjroberts**
https://ag-grid.atlassian.net/browse/AG-12978

### CODE REVIEW

File: packages/ag-charts-community/src/chart/data/processors.ts [hunk 1]
```diff
@@ -184,41 +184,48 @@ export const SORT_DOMAIN_GROUPS: ProcessorOutputPropertyDefinition<'sortedGroupD
...
+        const result = ((val ?? 0) * normaliseTo) / extent;
         if (result >= 0) {
             return Math.min(normaliseTo, result);
         }
         return Math.max(-normaliseTo, result);
     };
 
     return () => () => (columns: any[][], valueIndexes: number[], datumIndex: number) => {
-        const valuesExtent = [0, 0];
+        const extent = normaliseFindExtent(mode, columns, valueIndexes, datumIndex);
```
Comment (lsjroberts) [inline packages/ag-charts-community/src/chart/data/processors.ts:197]: Sonarqube was complaining about complexity. The actual difference is checking for nulls.

File: packages/ag-charts-community/src/chart/data/processors.ts [hunk 2]
```diff
@@ -184,41 +184,48 @@ export const SORT_DOMAIN_GROUPS: ProcessorOutputPropertyDefinition<'sortedGroupD
...
-                valuesExtent[valIdx] = Math.max(valuesExtent[valIdx], valueExtent);
-            }
-        }
-
-        const extent = Math.max(Math.abs(valuesExtent[0]), valuesExtent[1]);
-        for (const valueIdx of valueIndexes) {
-            const column = columns[valueIdx];
-            const value: number | number[] = column[datumIndex];
+            const value: null | number | number[] | (null | number)[] = column[datumIndex];
+            if (value == null) continue;
```
Comment (jacobp100) [inline packages/ag-charts-community/src/chart/data/processors.ts]: I think you want to set the `column[valueIdx]` to undefined or something here

**Review by jacobp100 — APPROVED** [review]

File: packages/ag-charts-community/src/chart/data/processors.ts [hunk 3]
```diff
@@ -184,41 +184,48 @@ export const SORT_DOMAIN_GROUPS: ProcessorOutputPropertyDefinition<'sortedGroupD
...
             column[datumIndex] =
                 typeof value === 'number' ? normalise(value, extent) : value.map((v) => normalise(v, extent));
         }
     };
 }
 
+function normaliseFindExtent(mode: 'sum' | 'range', columns: any[][], valueIndexes: number[], datumIndex: number) {
+    const valuesExtent = [0, 0];
```
Comment (jacobp100) [inline packages/ag-charts-community/src/chart/data/processors.ts:212]: Wonder if it'll be faster to store the sum, min, max and always update them, then figure out what they actually wanted at the end

Comment (jacobp100) [reply packages/ag-charts-community/src/chart/data/processors.ts:212]: (Ignore if you like though since this isn't a new issue)

**Pull request merged**